shpopen.c

```diff
@@ -818,7 +818,7 @@ int SHPAPI_CALL SHPRestoreSHX(const char *pszLayer, const char *pszAccess,
                 char szErrorMsg[200];
                 snprintf(szErrorMsg, sizeof(szErrorMsg),
                          "Error parsing .shp to restore .shx. "
-                         "Invalid record length = %u at record starting at "
+                         "Invalid record length = %d at record starting at "
                          "offset %u",
                          nSHPType, nCurrentSHPOffset);
```

Comment:
I believe the right fix would be:
```suggestion
                         "Invalid record length = %u at record starting at "
                         "offset %u",
                         nRecordLength, nCurrentSHPOffset);
```